cdk/kraken/src/docker.ts

```diff
@@ -53,11 +53,11 @@ export interface DockerPublishJobProps {
   cache?: boolean;
 
   /**
-   * If enabled, do not publish docker image to Docker Hub.
+   * If evaluates to true, do not publish docker image to Docker Hub.
    * Instead upload as an artifact.
    * @default false
    */
-  noPublish?: boolean;
+  noPublish?: boolean | string;
```

Comment:
I feel like `noPublish` having the possibility of being a `string` or a `boolean` is a little unintuitive.

Maybe we can change the comments that specifies `noPublish` is a "no-publish condition", or there's a better name for this variable.

Also, since it can be a string AND is optional, we can simplify the type to just be a `noPublish?: string;` and check if `noPublish` is undefined (truthy/falsey instead of true/false directly).

Reply:
Wouldn't this require a fair amount of refactoring in other places? Like we'd have to change `noPublish` to `"true"` or arbitrary truthy string value from `true` everywhere we specify it right?

IMO can stick to this but include better comments if ^ turns out to be true?

Reply:
Also stricter typing might be slightly more unintuitive but lead to fewer errors down the line if we're passing around config, accidentally passing around `undefined` could be problematic.

Reply:
Another consideration is having non-confusing typing. Either make a boolean disabling publishing, or make a variable specifying the conditions to disable publishing.

I took a look at existing code using a defined `noPublish` and the only places are a [test case]([docker.test.ts](https://github.com/pennlabs/infrastructure/blob/efd8251e7633214006c7d92c247079ee09c65b99/cdk/kraken/test/docker.test.ts)), which means this would not be a breaking change. Even if it was used in prod, for this change to apply to a product, we would need to bump `kraken`'s version in `package.json`.

Reply:
What would you lean towards for the default value of `integrationTests` if I want it to be enabled/disabled? Do you tink it's ok if convention is to set `integrationTests` to something like "true" as opposed to `true` in, for example, [this file](https://github.com/pennlabs/penn-clubs/blob/master/.github/cdk/main.ts)?

I'd be ok with that and just using truthy/falsey conditioning if you think it's fine!